internal/job/executor.go

```diff
@@ -768,6 +768,15 @@ func (e *Executor) PluginPhase(ctx context.Context) error {
 		return nil
 	}
 
+	// NOTE: these are method expressions: https://go.dev/ref/spec#MethodExpr
+	checkoutPluginFunc := (*Executor).checkoutPlugin
+	if experiments.IsEnabled(experiments.IsolatedPluginCheckout) {
+		if e.Debug {
+			e.shell.Commentf("Using isolated plugin checkout")
+		}
+		checkoutPluginFunc = (*Executor).checkoutPluginIsolated
+	}
```

Comment:
Definitely my favourite secret feature of Go. 

Since `checkoutPluginFunc` is always called with `e` as the first arg, maybe it would make more sense to capture `e.checkoutPlugin` instead?

```suggestion
	// NOTE: these are method expressions: https://go.dev/ref/spec#MethodExpr
	checkoutPluginFunc := e.checkoutPlugin
	if experiments.IsEnabled(experiments.IsolatedPluginCheckout) {
		if e.Debug {
			e.shell.Commentf("Using isolated plugin checkout")
		}
		checkoutPluginFunc = e.checkoutPluginIsolated
	}
```

and then:

```go
checkout, err := checkoutPluginFunc(ctx, p)
```

Reply:
Aww, then I guess they're no longer method expressions 😥

I've done this.